kernel/arch/riscv/debug.c

```diff
@@ -13,27 +13,191 @@
 
 #include <elf.h>
 
+static bool fp_is_valid(ulong fp, ulong sp)
+{
+   ulong low = sp + 2 * sizeof(void *);
+   ulong high = (sp + (KERNEL_STACK_SIZE - 1)) & ~(KERNEL_STACK_SIZE - 1);
+
+   if (!fp || fp < low || fp > high || fp & 0x7)
+      return false;
+   else
+      return true;
+}
+
+static size_t
+stackwalk32(void **frames,
```

Comment:
On x86, `stackwalk32()` was named like that to recall that's the (x86) 32-bit version of that function. Can you rename this function to something like `stackwalk_riscv()` so that we have less name clashing when searching etc. and so that it's clear that we have a stackwalk_* function for each architecture?

Reply:
fixed!